### PR TITLE
Refactoring function execution flow, ensuring that disabled functions can only be invoked with admin credentials

### LINKED ISSUE #2129: Handle "disabled" state for httpTrigger functions

### PR DESCRIPTION
**Pull request by fabiocav**
Resolves #2129

While the change could have been made as an enhancement to the existing route handler, I've split things up as I had described to allow for better separation of concerns and more flexibility when adding logic that requires information about an executing function.

### CODE REVIEW

File: src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs [hunk 1]
```diff
@@ -18,10 +18,8 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
 
         public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder builder, IApplicationLifetime applicationLifetime, Action<WebJobsRouteBuilder> routes)
         {
-            WebScriptHostManager hostManager = builder.ApplicationServices.GetService(typeof(WebScriptHostManager)) as WebScriptHostManager;
-
+            builder.UseMiddleware<FunctionInvocationMiddleware>();
```
Comment (brettsam) [inline src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs]: Can you elaborate on why this is the first middleware? I would have thought it'd be sitting after the routing one.

Comment (fabiocav) [reply src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs]: Once handled by the router, other middleware in the pipeline will not be executed, so this is registered first. You'll notice the actual middleware implementation handles the logic after executing the rest of the pipeline.

**Comment by fabiocav** [comment]
Meant to add a comment, but Alexey is adding a test that will validate this as part of his changes.

**Comment by fabiocav** [comment]
@brettsam; added some comments here

**Comment by fabiocav** [comment]
Merging this to unblock work depending on #2129 ; will address additional feedback as needed

**Pull request merged**